tasks/google/cloud/tasks_v2beta2/gapic/cloud_tasks_client.py

```diff
@@ -515,11 +546,17 @@ def update_queue(self,
             queue=queue,
             update_mask=update_mask,
         )
-
-        if hasattr(queue, 'name'):
-            routing_header = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
-                [('queue.name', queue.name)], )
-            metadata.append(routing_header)
+        if metadata is None:
+            metadata = []
+        metadata = list(metadata)
+        try:
+            routing_header = [('queue.name', queue.name)]
```

Comment:
Codegen error:  `queue` could be either a protobuf message (in which case it has a `name` attribute) or a dict (in which case it has a `name` key).  We need to handle the second case properly.